.github/workflows/build-plugin.yaml

```
@@ -0,0 +1,124 @@
+name: Build plugin
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v*'
+  pull_request:
+
+permissions: {}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        target:
+          - linux-x64-glibc
+          - darwin-x64-unknown
+          - win32-x64-unknown
+
+    name: Build plugin for ${{ matrix.target }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read # clone the repository
+      id-token: write # required to read secrets
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+      - run: yarn install --frozen-lockfile
```

Since we don't run tests here 🤔 
```suggestion
      - run: yarn install --production --frozen-lockfile
```

---

We actually need the dev dependencies for the `sign-plugin` call, but maybe two installs is better -- one for the prod deps (for the plugin), then reinstall for dev after we've packaged them...

---

We can't. The build fails:

```
$ tsc
Error: node_modules/@grpc/proto-loader/build/src/index.d.ts(18,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-amqplib/build/src/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-aws-sdk/build/src/aws-sdk.types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-fs/build/src/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-http/build/src/http.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-http/build/src/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-ioredis/build/src/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-kafkajs/build/src/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-redis-4/build/src/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/instrumentation-undici/build/src/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/otlp-exporter-base/build/src/platform/node/OTLPExporterNodeBase.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/otlp-exporter-base/build/src/platform/node/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/otlp-exporter-base/build/src/platform/node/util.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/sdk-node/node_modules/@opentelemetry/core/build/src/platform/node/performance.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/@opentelemetry/sdk-node/node_modules/@opentelemetry/core/build/src/platform/node/timer-util.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/chokidar/types/index.d.ts(3,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/Command.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/DataHandler.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/Redis.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/ScanStream.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/cluster/ClusterOptions.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/cluster/index.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/cluster/util.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/connectors/SentinelConnector/index.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/connectors/StandaloneConnector.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/ioredis/built/utils/RedisCommander.d.ts(6,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/pg-protocol/dist/index.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/pg-protocol/dist/messages.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/pg-protocol/dist/parser.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/pg-protocol/dist/parser.d.ts(2,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/poolpeteer/dist/Cluster.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/puppeteer-cluster/dist/Cluster.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/puppeteer/lib/types.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/winston-transport/index.d.ts(6,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/winston/index.d.ts(4,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/winston/lib/winston/config/index.d.ts(4,23): error TS2688: Cannot find type definition file for 'node'.
Error: node_modules/winston/lib/winston/transports/index.d.ts(4,23): error TS2688: Cannot find type definition file for 'node'.
```